sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3751,4 +3751,14 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
         "supported" -> "constant expressions"),
       cause = cause)
   }
+
+  def unsupportedCorrelatedSubqueryExpressionInJoinConditionError(
```

Comment:
In my experience it's better not to add items at the end because if we need to backport/cherry-pick it creates tons of merge conflicts - better to add in the middle, perhaps near related confs in UNSUPPORTED_SUBQUERY_EXPRESSION_CATEGORY

Reply:
Fixed!